spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsMessageDrivenEndpoint.java

```diff
@@ -35,12 +36,33 @@
 
 	private final AbstractMessageListenerContainer listenerContainer;
 
+	private final boolean externalContainer;
+
 	private final ChannelPublishingJmsMessageListener listener;
-	
+
 	private volatile String sessionAcknowledgeMode;
-	
+
+	/**
+	 * Construct an instance with an externally configured container.
+	 * @param listenerContainer the container.
+	 * @param listener the listener.
+	 */
 	public JmsMessageDrivenEndpoint(AbstractMessageListenerContainer listenerContainer,
 			ChannelPublishingJmsMessageListener listener) {
+		this(listenerContainer, listener, true);
+	}
+
+	/**
+	 * Construct an instance with an argument indicating whether the container's ack mode should
+	 * be overridden with {@link #setSessionAcknowledgeMode(String) sessionAcknowledgeMode}, default
+	 * 'transacted'.
+	 * @param listenerContainer the container.
+	 * @param listener the listener.
+	 * @param externalContainer true if the container is externally configured and should not have its ackmode
+	 * coerced when no sessionAcknowledgeMode was supplied.
+	 */
+	public JmsMessageDrivenEndpoint(AbstractMessageListenerContainer listenerContainer,
+			ChannelPublishingJmsMessageListener listener, boolean externalContainer) {
```

Comment:
This ctor looks strange from JavaConfig perspective, but what is the choice for XML...
I know! 
The `private` modification for this works well with the `BeanDefinition` :smile: